lib/cocoapods-core/source.rb

```diff
@@ -326,6 +369,18 @@ def specs_dir
       end
     end
 
+    def git_commit_hash
+      (`git rev-parse HEAD` || '').strip
+    end
+
+    def update_git_repo
```

Comment:
I don't think this is needed because it's output is handled and streamed in `update` method above. 
Look at the lines between 292-300. It uses CoreUI which is handled by CocoaPods by default in `user_interface.rb` as far as I understand.